test/models/trip_test.dart

```diff
@@ -0,0 +1,369 @@
+    });
+    test('trip overlaps the other', () {
+      final Trip trip1 = (randomTripFactory() as TripFactory).generateFake(
+          startTime: DateTime.now().subtract(const Duration(hours: 6)),
+          endTime: DateTime.now().subtract(const Duration(hours: 2)));
+      final Trip trip2 = (randomTripFactory() as TripFactory).generateFake(
+          startTime: DateTime.now().subtract(const Duration(hours: 3)),
+          endTime: DateTime.now().add(const Duration(hours: 2)));
+      expect(trip1.overlapsWith(trip2), true);
+    });
+  });
+  group('Trip.overlapswithRange', () {
+    test('trip is after range', () {
+      final DateTimeRange range = DateTimeRange(
+        start: DateTime.now().subtract(const Duration(hours: 6)),
+        end: DateTime.now().subtract(const Duration(hours: 3)),
+      );
+      final Trip trip = (randomTripFactory() as TripFactory).generateFake(
+          startTime: DateTime.now().add(const Duration(hours: 4)),
+          endTime: DateTime.now().add(const Duration(hours: 2)));
+      expect(trip.overlapsWithTimeRange(range), false);
+    });
+    test('trip overlaps with range', () {
+      final DateTimeRange range = DateTimeRange(
+        start: DateTime.now().add(const Duration(hours: 3)),
+        end: DateTime.now().add(const Duration(hours: 6)),
+      );
+      final Trip trip = (randomTripFactory() as TripFactory).generateFake(
+          startTime: DateTime.now().add(const Duration(hours: 2)),
+          endTime: DateTime.now().add(const Duration(hours: 4)));
+      expect(trip.overlapsWithTimeRange(range), true);
+    });
+  });
+  group('Trip.shouldShowInListView', () {
+    test('returns true if trip should show in ListView', () {
+      final Trip ride = RideFactory().generateFake(
+        status: RideStatus.approved,
+        startTime: DateTime.now().add(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(ride.shouldShowInListView(past: false), true);
+      final Trip drive = DriveFactory().generateFake(
+        cancelled: false,
+        startTime: DateTime.now().add(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(drive.shouldShowInListView(past: false), true);
+    });
+    test('returns false for trip with hideInListView', () {
+      final Trip ride = RideFactory().generateFake(
+        hideInListView: true,
+        startTime: DateTime.now().add(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(ride.shouldShowInListView(past: false), false);
+      final Trip drive = DriveFactory().generateFake(
+        hideInListView: true,
+        startTime: DateTime.now().add(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(drive.shouldShowInListView(past: false), false);
+    });
+    test('returns true for past finsihed trip', () {
```

Comment:
```suggestion
    test('returns true for past finished trip', () {
```

Please use a code spell checker :) I'm not going to comment on spelling mistakes anymore from here on out